### PR TITLE
support LIKE comparator (full text match) in Qdrant

### PR DESCRIPTION
**Pull request by xieqihui**
**Description:** 
Support [Qdrant full text match filtering](https://qdrant.tech/documentation/concepts/filtering/#full-text-match) by adding Comparator.LIKE to QdrantTranslator.

### CODE REVIEW

File: libs/langchain/langchain/retrievers/self_query/qdrant.py [hunk 1]
```diff
@@ -69,6 +70,10 @@ def visit_comparison(self, comparison: Comparison) -> rest.FieldCondition:
             return rest.FieldCondition(
                 key=attribute, match=rest.MatchValue(value=comparison.value)
             )
+        if comparison.comparator == Comparator.LIKE:
+            return rest.FieldCondition(
+                key=attribute, match=rest.MatchText(text=comparison.value)
```
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/self_query/qdrant.py:75]: what's the difference between MatchValue and MatchText?

**Review by baskaryan — COMMENTED** [review]
cc @kacperlukawski

File: libs/langchain/langchain/retrievers/self_query/qdrant.py (hunk 1 above)
Comment (xieqihui) [reply libs/langchain/langchain/retrievers/self_query/qdrant.py:75]: MatchValue is exact match, while MatchText supports phrase match (provided that the full-text index is configured for the target field in Qdrant db). MatchText can be used in scenarios where fuzzy search is needed. 

For example, if the target keyword is "LangChain LLM": 
- With MatchValue, only query == "LangChain LLM" can match the target. 
- With MatchText, all queries with substring like "langchain llm", "langchain", "LangChain LLM" may match the target, depending on how you configure the full-text index. 

If full-text index is not configured for the field in the db, MatchText will work the same as MatchValue.

**Review by kacperlukawski — APPROVED** [review]
LGTM, I was unaware of that comparator, but that's a great addition! Thanks, @xieqihui!

File: libs/langchain/langchain/retrievers/self_query/qdrant.py (hunk 1 above)
Comment (xieqihui) [reply libs/langchain/langchain/retrievers/self_query/qdrant.py:75]: Can this PR be merged?

**Comment by kacperlukawski** [comment]
@baskaryan Could we merge this PR?

**Comment by xieqihui** [comment]
@baskaryan Could we merge this PR please?

**Comment by xieqihui** [comment]
hi @hwchase17 ，may I know the reason why this PR is closed? Could you please give some suggestions?

**Comment by xieqihui** [comment]
hi @hwchase17 ，may I know the reason why this PR is closed? Could you please give some suggestions?
I have been using this feature in my production project. It's quite useful to support fuzzy search with Qdrant.

**Comment by xieqihui** [comment]
hi @hwchase17 ，may I know the reason why this PR is closed? Could you please give some suggestions?
I have been using this feature in my production project. It's quite useful to support fuzzy search with Qdrant. Really looking forward to have it in Langchain master so that I don't need to keep a diverged code.

**Review by Anush008 — APPROVED** [review]
Thank you @xieqihui.

**Pull request merged**